src/context.ts

```diff
@@ -76,7 +76,10 @@ export class PageContext {
     debug.pages('add', path)
     for (const p of toArray(path)) {
       const pageDirPath = slash(resolve(this.root, pageDir.dir))
-      const route = slash(join(pageDir.baseRoute, p.replace(`${pageDirPath}/`, '').replace(extname(p), '')))
+      const extension = this.options.extensions.find(e => p.endsWith(e))
+      if (!extension) continue
+
+      const route = slash(join(pageDir.baseRoute, p.replace(`${pageDirPath}/`, '').replace(extension, '')))
```

Comment:
```suggestion
      const extension = this.options.extensions.find(ext => p.endsWith(`.${ext}`))
      if (!extension) continue

      const route = slash(join(pageDir.baseRoute, p.replace(`${pageDirPath}/`, '').replace(`.${extension}`, '')))
```

**Required**: we should remove extension including `.` from route path

Reply:
Oh sorry, I didn't make sure to work properly after fixing it, thanks.